cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,869 @@
+    /**
+     * Create a watch on {@code KafkaRebalance} in the given {@code watchNamespaceOrWildcard}.
+     *
+     * @param watchNamespaceOrWildcard The namespace to watch, or "*" to watch all namespaces.
+     * @return A future which completes when the watch has been set up.
+     */
+    public Future<Void> createRebalanceWatch(String watchNamespaceOrWildcard) {
+
+        return Util.async(this.vertx, () -> {
+            this.kafkaRebalanceOperator.watch(watchNamespaceOrWildcard, new Watcher<KafkaRebalance>() {
+                @Override
+                public void eventReceived(Action action, KafkaRebalance kafkaRebalance) {
+                    Reconciliation reconciliation = new Reconciliation("kafkarebalance-watch", kafkaRebalance.getKind(),
+                            kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName());
+
+                    log.debug("{}: EventReceived {} on {} with status [{}] and {}={}", reconciliation, action,
+                            kafkaRebalance.getMetadata().getName(),
+                            kafkaRebalance.getStatus() != null ? kafkaRebalance.getStatus().getConditions().get(0).getType() : null,
+                            ANNO_STRIMZI_IO_REBALANCE, rawRebalanceAnnotation(kafkaRebalance));
+
+                    withLock(reconciliation, LOCK_TIMEOUT_MS,
+                        () -> reconcileRebalance(reconciliation, action == Action.DELETED ? null : kafkaRebalance));
+                }
+
+                @Override
+                public void onClose(KubernetesClientException e) {
+                    if (e != null) {
+                        throw e;
+                    }
+                }
+
+            });
+            return null;
+        });
+    }
+
+    private static Future<KafkaRebalance> updateStatus(KafkaRebalance clusterRebalance,
+                                                      KafkaRebalanceStatus desiredStatus,
+                                                      CrdOperator<KubernetesClient, KafkaRebalance, KafkaRebalanceList, DoneableKafkaRebalance> clusterRebalanceOperations,
+                                                      Throwable e) {
+        if (e != null) {
+            StatusUtils.setStatusConditionAndObservedGeneration(clusterRebalance, desiredStatus, e);
+        } else if (desiredStatus.getConditions() != null && desiredStatus.getConditions().get(0).getType() != null) {
+            StatusUtils.setStatusConditionAndObservedGeneration(clusterRebalance, desiredStatus,
+                    desiredStatus.getConditions().get(0).getType());
+        } else {
+            throw new IllegalArgumentException("Status related exception and type cannot be both null");
+        }
+        StatusDiff diff = new StatusDiff(clusterRebalance.getStatus(), desiredStatus);
+        if (!diff.isEmpty()) {
+            KafkaRebalance copy = new KafkaRebalanceBuilder(clusterRebalance).build();
+            copy.setStatus(desiredStatus);
```

Comment:
Surely the idiomatic way to do this is to set the status in the builder, rather than build and then set? Then you can inline the whole builder expression and you only need one line where you currently have 3.